src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -47,14 +47,23 @@ class ValidateActor(wdlSource: WdlSource, wdlJson: WdlJson)
 
   private def validateWorkflow(sentBy: ActorRef): Unit = {
     logger.info(s"$tag for $sentBy")
-    val futureValidation: Future[Unit] = for {
+    val futureValidation: Future[NamespaceWithWorkflow] = for {
       namespaceWithWorkflow <- Future(NamespaceWithWorkflow.load(wdlSource))
-      inputs <- Future(wdlJson.parseJson).map(_.asJsObject.fields)
-      coercedInputs <- Future.fromTry(namespaceWithWorkflow.coerceRawInputs(inputs))
       runtime = namespaceWithWorkflow.workflow.calls foreach { _.toRuntimeAttributes }
-    } yield () // Validate that the future run and return `Success[Unit]` aka (), or `Failure[Exception]`
 
-    futureValidation onComplete {
+    } yield namespaceWithWorkflow
+
+    val futureValidationIncludingInputs: Future[Any] = workflowInputs match {
```

Comment:
Hmm I liked the `Future[Unit]` better for not leaking information or suggesting that the information being returned would be useful to a caller.